### ISSUE TITLE
Wrong testing accuracy for testing data

### ISSUE THREAD

**Issue by odutfoy** [iit-cs579/main#103]
Hello,

I get a 0.5 accuracy on the testing data instead of 0.78. I'm not the only one with this problem and I wondered if someone had it too and how they manage to fix it.

Thank you

**Comment by prateekagrawal06** [comment]
Hi...i was having the same issue...but now m getting it right.

Problem might be the way u read the files. You should use the input parameter of the countVectorizer and pass the list of filenames in the vec.fit_transform() function. I think this should resolve it for you.

**Comment by odutfoy** [comment]
It did indeed, thank you very much

**Comment by goutkannan** [comment]
Hi,

In CountVector, I have  Input parameter as list of filenames and in fit_transform() I have passed the , all the strings present in each file. 

I still get test accuracy as 0.5. Am I missing anything here..? 

Thnks

**Comment by odutfoy** [comment]
You have to put the name of the files in fit_transform(), not the string they contain

**Comment by jaydeeppatel1701** [comment]
when i tried to use filenames in fit_transform, i am getting only 405 features.  
do i have to pass filenames directly or some other way?

**Comment by bpraveen92** [comment]
The input type has to be given as 'filename' as you pass a list of filenames to the vectorizer object. So double check what exact parameter you've passed in Countvectorizer.

**Comment by hinagarg** [comment]
u have to pass it by using input parameter in CountVectorizer

**Comment by jaydeeppatel1701** [comment]
i passed it in CountVectorizer function input=filenames. But for fit_transform function i am reading each file and storing these strings into list. if i pass filenames in fit_transform then i am not able to fetch 10710 features

**Comment by odutfoy** [comment]
Don't open the files. Store the names of the files in your list instead
Le 5 nov. 2015 21:38, "jaydeeppatel1701" <notifications@github.com> a
écrit :

**Comment by goutkannan** [comment]
@jaydeeppatel1701 , It is input='filename' within single quotation,it is not the parameter filenames, we have to give a keyword 'filename'

**Comment by jaydeeppatel1701** [comment]
Thanks @goutkannan, @hinagarg @bpraveen92 @prateekagrawal06  and @odutfoy for your insights. You saved the day.
